src/qibolab/instruments/zhinst.py

```diff
@@ -1218,8 +1221,20 @@ def sweep_recursion_nt(self, qubits, couplers, options, exp, exp_calib):
             else:
                 self.define_exp(qubits, couplers, options, exp, exp_calib)
 
-    def play_sequences(self, qubits, sequence, options):
-        pass
+    def split_batches(self, sequences):
+        batch_measurements, batch = 0, []
+        for sequence in sequences:
+            nmeasurements = len(sequence.ro_pulses)
+            if nmeasurements + batch_measurements > MAX_MEASUREMENTS:
+                yield batch
+                batch_measurements, batch = nmeasurements, [sequence]
+            else:
+                batch.append(sequence)
+                batch_measurements += nmeasurements
+        yield batch
```

Comment:
We could consider promoting this to the general implementation, if we could understand how things the `MAX_MEASUREMENTS` is scaling with the qubits (however, we can even have both the Qblox style and this has standalone functions, since they do not require `self` at all, and use or the other - each instrument will have to just register the `max_measurements`, that could even be a customizable class variable).